### PR TITLE
vpc,event: fix events for createVpc

### LINKED ISSUE #8496: Event state  is incorrect when  vpc network is created
ISSUE TYPE

Bug

COMPONENT NAME

Component: API/ UI

CLOUDSTACK VERSION

Cloudstack version 4.18, 

SUMMARY

Event is incorrectly displayed when vpc network is created

Steps to reproduce the issue 

1. Navigate to Network > VPC

2. Create a vpc 

5. Only the following  events generated for the vpc 

Screenshot 2024-01-11 at 11 44 56 AM

```
*************************** 325. row ***************************
           id: 325
         uuid: 1c4a28b8-fa6c-484a-9297-aa16ee3453f4
         type: VPC.CREATE
        state: Created
  description: Successfully created entity for creating vpc
      user_id: 2
   account_id: 2
    domain_id: 1
  resource_id: 2
resource_type: Vpc
      created: 2024-01-11 06:11:51
        level: INFO
     start_id: 0
   parameters: NULL
     archived: 0
      display: 1
*************************** 326. row ***************************
           id: 326
         uuid: 83458427-e905-4b8e-afed-032141ee7134
         type: VPC.CREATE
        state: Scheduled
  description: creating VPC. Id: 2
      user_id: 2
   account_id: 2
    domain_id: 1
  resource_id: 2
resource_type: Vpc
      created: 2024-01-11 06:11:51
        level: INFO
     start_id: 325
   parameters: NULL
     archived: 0
      display: 1

```

Expected behaviour 

The state of the VPC.create event should be completed

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Fixes #8496

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### Screenshots (if appropriate):
image

### CODE REVIEW

File: server/src/main/java/com/cloud/network/vpc/VpcManagerImpl.java [hunk 1]
```diff
@@ -1677,6 +1677,17 @@ public boolean startVpc(final long vpcId, final boolean destroyOnFailure) throws
...
 
+
+    @Override
+    @ActionEvent(eventType = EventTypes.EVENT_VPC_CREATE, eventDescription = "creating vpc", async = true)
+    public void startVpc(final CreateVPCCmd cmd) throws ConcurrentOperationException, ResourceUnavailableException, InsufficientCapacityException {
+        if (!cmd.isStart()) {
+            s_logger.debug("Not starting VPC as " + ApiConstants.START + "=false was passed to the API");
+            return;
+        }
+        startVpc(cmd.getEntityId(), false);
```
Comment (sureshanaparti) [inline server/src/main/java/com/cloud/network/vpc/VpcManagerImpl.java:1688]: earlier call has destroyOnFailure set to true, here it is false. does it break any existing behavior?

Comment (shwstppr) [inline server/src/main/java/com/cloud/network/vpc/VpcManagerImpl.java:1688]:
```suggestion
        startVpc(cmd.getEntityId(), true);
```

Comment (shwstppr) [reply server/src/main/java/com/cloud/network/vpc/VpcManagerImpl.java]: updated

**Comment by shwstppr** [comment]
@blueorangutan package

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by sureshanaparti** [comment]
@blueorangutan test

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by shwstppr** [comment]
@blueorangutan package

**Review by kiranchavala — APPROVED** [review]
LGTM, VPC events are properly generated 

Before fix

beforevpc(event)


After fix

Screenshot 2024-06-12 at 12 50 46 PM

**Pull request merged**